astropy/utils/masked/function_helpers.py

```diff
@@ -1076,9 +1083,9 @@ def array_str(a, max_line_width=None, precision=None, suppress_small=None):
     # code turns the masked array scalar into a regular array scalar.
     # By going through MaskedFormat, we can replace the string as needed.
     if a.shape == () and a.dtype.names is None:
-        return MaskedFormat(_array_str_scalar)(a)
-
-    return array2string(a, max_line_width, precision, suppress_small, " ", "")
+        return MaskedFormat(_array_str_scalar)(a), None, None
+    else:
+        return array2string(a, max_line_width, precision, suppress_small, " ", "")
```

Comment:
This also needs the `, None, None`

Reply:
actually it doesn't, `array2string` already returns a correctly-shaped tuple ! (I got bitten too)

Reply:
Ah, right, because it calls the helper not the numpy function!